packages/netlify-cms-widget-relation/src/RelationControl.js

```diff
@@ -165,26 +165,25 @@ export default class RelationControl extends React.Component {
 
   handleChange = selectedOption => {
     const { onChange, field } = this.props;
-    let value;
-    let metadata;
-
-    if (Array.isArray(selectedOption)) {
-      this.setState({ initialOptions: selectedOption.filter(Boolean) });
-      value = selectedOption.map(optionToString);
-      metadata =
-        (!isEmpty(selectedOption) && {
+
+    if (this.isMultiple()) {
```

Comment:
See https://github.com/JedWatson/react-select/releases/tag/react-select%403.0.3 under `Normalized Values`.
Clearing the select no longer returns an empty array and now returns `null`. This means we can't use `Array.isArray(selectedOption)` to determine if this is a multiple field or not.